class.jetpack-data.php

```diff
@@ -26,12 +31,43 @@ public static function get_access_token( $user_id = false ) {
 			if ( $user_id != $token_chunks[2] ) {
 				return false;
 			}
-			$token = "{$token_chunks[0]}.{$token_chunks[1]}";
+			$tokens = array( "{$token_chunks[0]}.{$token_chunks[1]}" );
 		} else {
+			$tokens = Jetpack_Constants::is_defined( 'JETPACK_BLOG_TOKEN' ) && ';stored;' !== $token_key
+				? explode( ',', Jetpack_Constants::get_constant( 'JETPACK_BLOG_TOKEN' ) )
+				: array();
+
 			$token = Jetpack_Options::get_option( 'blog_token' );
-			if ( empty( $token ) ) {
+			if ( empty( $token ) && empty( $tokens ) ) {
+				return false;
+			}
+
+			$tokens[] = $token;
```

Comment:
We're interested in the first token that matches the provided `$token_key`, which is usually the first one (since most sites will only have one token), but sometimes not.